### PR TITLE
MINOR: Fix setting of ACLs and ZK shutdown in test harnesses

### PR DESCRIPTION
**Pull request by ijuma**
I found both issues while investigating the issue described in PR #1425.

### CODE REVIEW

**Comment by ijuma** [comment]
Review by @gwenshap or @junrao.

**Comment by harshach** [comment]
+1

File: core/src/test/scala/unit/kafka/zk/EmbeddedZookeeper.scala [hunk 1]
```diff
@@ -39,6 +39,16 @@ class EmbeddedZookeeper() {
   def shutdown() {
     CoreUtils.swallow(zookeeper.shutdown())
     CoreUtils.swallow(factory.shutdown())
+
+    def isDown(): Boolean = {
```
Comment (junrao) [inline core/src/test/scala/unit/kafka/zk/EmbeddedZookeeper.scala:43]: There are a few tests that only extend ZooKeeperTestHarness. Do we need to ensure that ZK is down when shutting down ZooKeeperTestHarness?

**Comment by junrao** [comment]
Thanks for the patch. LGTM. Just left a minor comment.

File: core/src/test/scala/unit/kafka/zk/EmbeddedZookeeper.scala (hunk 1 above)
Comment (ijuma) [inline core/src/test/scala/unit/kafka/zk/EmbeddedZookeeper.scala:43]: `ZooKeeperTestHarness.tearDown` calls this method:

``` scala
if (zookeeper != null)
        CoreUtils.swallow(zookeeper.shutdown())
```

Isn't that good enough?

Comment (junrao) [inline core/src/test/scala/unit/kafka/zk/EmbeddedZookeeper.scala:43]: Yes, you are right. The code looks good then.

**Comment by ijuma** [comment]
Thank you for the reviews, merged to trunk and 0.10.0 branches (KAFKA-3396 which would be good to backport requires these changes for the tests to work).

**Pull request closed without merging**